internal/otelcollector/config/metric/gateway/receivers.go

```diff
@@ -0,0 +1,46 @@
+package gateway
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/ports"
+)
+
+func makeReceiversConfig() Receivers {
+	return Receivers{
+		OTLP: config.OTLPReceiver{
+			Protocols: config.ReceiverProtocols{
+				HTTP: config.Endpoint{
+					Endpoint: fmt.Sprintf("${%s}:%d", config.EnvVarCurrentPodIP, ports.OTLPHTTP),
+				},
+				GRPC: config.Endpoint{
+					Endpoint: fmt.Sprintf("${%s}:%d", config.EnvVarCurrentPodIP, ports.OTLPGRPC),
+				},
+			},
+		},
+	}
+}
+
+func makeSingletonKymaStatsReceiverCreatorConfig(gatewayNamespace string) *SingletonKymaStatsReceiverCreator {
+	return &SingletonKymaStatsReceiverCreator{
+		AuthType: "serviceAccount",
+		LeaderElection: LeaderElection{
+			LeaseName:      "telemetry-metric-gateway-kymastats",
```

Comment:
Should not we pass the name of the lease as a parameter?

Reply:
We agreed that since this is the only place so far where the lease name is used, so there is no need now to pass it as a parameter